subcommands/keys/ca_revoke_device_ca.go

```diff
@@ -0,0 +1,83 @@
+		Run:   doRevokeDeviceCa,
+		Args:  cobra.ExactArgs(1),
+		Long: `Revoke device CA from the list of CAs allowed to issue device client certificates.
+
+When the online or local device CA is revoked:
+- It is no longer possible to register new devices with client certificates issued by this device CA.
+- Existing devices with client certificates issued by this device CA can no longer connect to your FoundriesFactory.
+
+You may later re-add a revoked device CA using "keys ca update", if you still keep its certificate stored somewhere.
+Once you do it, devices with client certificates issued by this device CA may connect to your FoundriesFactory again.`,
+	}
+	caCmd.AddCommand(cmd)
+	cmd.Flags().BoolP("dry-run", "", false,
+		"Do not revoke the certificate, but instead show a generated CRL that will be uploaded to the server.")
+	cmd.Flags().BoolP("pretty", "", false,
+		"Can be used with dry-run to show the generated CRL in a pretty format.")
+	cmd.Flags().StringArrayP("ca-file", "", nil,
+		"A file name of the device CA to revoke. Can be used several times to revoke several device CAs")
+	cmd.Flags().StringArrayP("ca-serial", "", nil,
+		"A serial number (base 10) of the device CA to revoke. Can be used several times to revoke several device CAs")
+	_ = cmd.MarkFlagFilename("ca-file")
+	// HSM variables defined in ca_create.go
+	cmd.Flags().StringVarP(&hsmModule, "hsm-module", "", "", "Load a root CA key from a PKCS#11 compatible HSM using this module")
+	cmd.Flags().StringVarP(&hsmPin, "hsm-pin", "", "", "The PKCS#11 PIN to log into the HSM")
+	cmd.Flags().StringVarP(&hsmTokenLabel, "hsm-token-label", "", "", "The label of the HSM token containing the root CA key")
+}
+
+func doRevokeDeviceCa(cmd *cobra.Command, args []string) {
+	factory := viper.GetString("factory")
+	// TODO: Implement the --dry-run and --pretty arguments
+	// dryRun, _ := cmd.Flags().GetBool("dry-run")
+	// pretty, _ := cmd.Flags().GetBool("pretty")
+	caFiles, _ := cmd.Flags().GetStringArray("ca-files")
+	caSerials, _ := cmd.Flags().GetStringArray("ca-serials")
+
+	if len(caFiles)+len(caSerials) == 0 {
+		subcommands.DieNotNil(errors.New("At least one of --ca-file or --ca-serial must be provided"))
+	}
+
+	subcommands.DieNotNil(os.Chdir(args[0]))
+	hsm, err := x509.ValidateHsmArgs(
+		hsmModule, hsmPin, hsmTokenLabel, "--hsm-module", "--hsm-pin", "--hsm-token-label")
+	subcommands.DieNotNil(err)
+	x509.InitHsm(hsm)
+
+	fmt.Println("Generating Cartificate Revokation List")
+	toRevoke := make(map[string]int, len(caSerials)+len(caFiles))
+	for _, serial := range caSerials {
+		num := new(big.Int)
+		if _, ok := num.SetString(serial, 10); !ok {
+			subcommands.DieNotNil(fmt.Errorf("Value %s is not a valid base 10 serial", serial))
+		}
+		toRevoke[serial] = x509.CrlCaRevoke
+	}
+	for _, filename := range caFiles {
+		ca := x509.LoadCertFromFile(filename)
+		toRevoke[ca.SerialNumber.Text(10)] = x509.CrlCaRevoke
```

Comment:
Why is it specifically 10? And always 10?

Reply:
It is base 10, and it will always be like this.

The reason: old versions of Fioctl picked base 10 as an arbitrary value to print the certificate serial number.
So, I leave it that way, and set it in stone.